internal/core/data/init.go

```diff
@@ -62,14 +64,15 @@ func (b *Bootstrap) BootstrapHandler(ctx context.Context, wg *sync.WaitGroup, st
 	// For Redis Streams MessageBus, we reuse the Redis instance running for the DB, which may have a password,
 	// so we need to get and use the DB credentials for the MessageBus connection.
 	if configuration.MessageQueue.Type == "redisstreams" {
-		credentials, err := container.CredentialsProviderFrom(dic.Get).GetDatabaseCredentials(configuration.Databases["Primary"])
+		secretProvider := container.SecretProviderFrom(dic.Get)
+		credentials, err := secretProvider.GetSecrets(configuration.Databases["Primary"].Type)
 		if err != nil {
 			lc.Error(fmt.Sprintf("Error getting DB creds for RedisStreams: %s", err.Error()))
 			return false
 		}
 
 		lc.Info("DB Credentials set for using Redis Streams")
-		configuration.MessageQueue.Optional["Password"] = credentials.Password
+		configuration.MessageQueue.Optional["Password"] = credentials[secret.PasswordKey]
```

Comment:
if we change PasswordKey to PascalCase, it could also be applied to `configuration.MessageQueue.Optional[secret.PasswordKey]` here

Reply:
I like where you are going with this, but are dependent of existing code that is creating them with lowercase. Changing that code is out of scope for this PR.
https://github.com/edgexfoundry/edgex-go/blob/master/internal/security/secretstore/password.go#L66